src/main/scala/cromwell/webservice/CallCachingParameters.scala

```diff
@@ -0,0 +1,137 @@
+package cromwell.webservice
+
+import cromwell.engine.WorkflowId
+import cromwell.engine.db.DataAccess._
+import cromwell.engine.db.{DataAccess, ExecutionDatabaseKey}
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+import scalaz.Scalaz._
+import scalaz.{Failure, Success, ValidationNel}
+
+case class CallCachingParameters private(workflowId: WorkflowId, callKey: Option[ExecutionDatabaseKey], allow: Boolean)
+
+object CallCachingParameters {
+
+  private val CallNamePattern =
+    """
+      (?x)                               # Turn on comments and whitespace insensitivity.
+
+      (                                  # Begin outer capturing group for FQN.
+
+        (?:[a-zA-Z][a-zA-Z0-9_]*)        #   Inner noncapturing group for top-level workflow name. This is the WDL
+                                         #   identifier naming pattern of an initial alpha character followed by zero
+                                         #   or more alphanumeric or underscore characters.
+
+        (?:\.[a-zA-Z][a-zA-Z0-9_]*){1}   #   Inner noncapturing group for call name, a literal dot followed by a WDL
+                                         #   identifier.  Currently this is quantified to {1} since the call name is
+                                         #   mandatory and nested workflows are not supported.  This could be changed
+                                         #   to + or a different quantifier if these assumptions change.
+
+      )                                  # End outer capturing group for FQN.
+
+
+      (?:                                # Begin outer noncapturing group for shard.
+        \.                               #   Literal dot.
+        (\d+)                            #   Captured shard digits.
+      )?                                 # End outer optional noncapturing group for shard.
+    """.trim.r                           // The trim is necessary as (?x) must be at the beginning of the regex.
```

Comment:
It'd be a shame if this was only ever used as a private val in the CallCaching endpoint. Can we generalise and extract this regex somewhere more widely useable?